mllib/src/main/scala/org/apache/spark/mllib/feature/IDF.scala

```diff
@@ -218,7 +218,7 @@ private object IDFModel {
           newValues(k) = values(k) * idf(indices(k))
           k += 1
         }
-        Vectors.sparse(n, indices, newValues)
+        Vectors.sparse(n, indices, newValues).toSparse
```

Comment:
@dbtsai what do you think about it as it looks like you made the test?
https://github.com/apache/spark/commit/b5d17ef10e2509d9886c660945449a89750c8116
Seems like it should be valid for sparse vectors to ignore 0 entries.